### PR TITLE
feat: add CLJS project to `create-tauri-app`, fix: #2822

### PR DESCRIPTION
**Pull request by wildwestrom**
I really wanted a way to easily bootstrap a tauri project using ClojureScript. Luckily it wasn't too hard to modify one of the recipes in `create-tauri-app`.

Mentioned in issue: #2822

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: tooling/create-tauri-app/package.json
Comment (amrbashir) [inline tooling/create-tauri-app/package.json]: why is this needed ?

File: tooling/create-tauri-app/src/recipes/cljs.ts
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/cljs.ts:21]: you should also add `beforeDevCommand` and `beforeBuildCommand` configuratons.

File: tooling/create-tauri-app/src/recipes/cljs.ts
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/cljs.ts]: The url in the name should link directly to the template site/github repo, check `svelte.ts` for example.

File: tooling/create-tauri-app/src/recipes/cljs.ts
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/cljs.ts]: I am not a coljurescript user but I quickly checked `create-cljs-app` and the [templates](https://clojurescript.org/guides/project-templates) mentioned on coljurescript official site and they all seem a bit outdated and not maintained. I think we should go with [`shadow-cljs`](https://github.com/thheller/shadow-cljs#quick-start) (`yarn create cljs-project`) instead, it is well maintained and comes with a sensible default features.

File: tooling/create-tauri-app/src/recipes/cljs.ts
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/cljs.ts]: Why though ? The whole purpose of `create-tauri-app` is people won't have to do this because we pre-configured `beforeDevCommand` and `beforeBuildCommand` for them.

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks for your contribution. Just a few changes requested and we should be good to go.

File: tooling/create-tauri-app/package.json
Comment (wildwestrom) [reply tooling/create-tauri-app/package.json]: it isn't really needed, it was just something missing when I tried to run `lockfile-lint`. If eslint is included, why not lockfile-lint?

Comment (amrbashir) [reply tooling/create-tauri-app/package.json]: I see, this should be in a different PR though, and probably with a package.json script that runs `lockfile-lint` and add it to our CI.

File: tooling/create-tauri-app/src/recipes/cljs.ts
Comment (wildwestrom) [reply tooling/create-tauri-app/src/recipes/cljs.ts]: Unfortunately, this project, [`create-cljs-project`](https://github.com/thheller/shadow-cljs/tree/master/packages/create-cljs-project) is currently a stub. Here it is on [npmjs.com](https://www.npmjs.com/package/create-cljs-project).

Comment (amrbashir) [reply tooling/create-tauri-app/src/recipes/cljs.ts]: I see, it is fine then.

File: tooling/create-tauri-app/src/recipes/cljs.ts
Comment (wildwestrom) [reply tooling/create-tauri-app/src/recipes/cljs.ts]: Seems reasonable to me. I just don't know how to overwrite the scripts from `create-cljs-app` as it has `start` and `build` already.

Comment (amrbashir) [reply tooling/create-tauri-app/src/recipes/cljs.ts]: you don't need to overwrite them, you need to specify them in `beforeDevCommand` and `beforeBuildCommand` (see https://github.com/tauri-apps/tauri/pull/2843#discussion_r744153972) and `tauri dev` or `tauri build` will run them automatically.

Comment (wildwestrom) [reply tooling/create-tauri-app/src/recipes/cljs.ts]: Ah ok, I didn't understand that's what `before(Dev|Build)Command` was doing.

File: tooling/create-tauri-app/src/recipes/cljs.ts
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/cljs.ts]: good catch, should probably also re-install dependencies after this 
```suggestion
      /* `create-cljs-app` has both a `package-lock.json` and a `yarn.lock`
         I think it's a good idea to get rid of conflicting lockfiles. */
      const npmLock = join(cwd, cfg.appName, 'package-lock.json')
      if (existsSync(npmLock)) rmSync(npmLock)
      await shell('yarn', ['install'], { cwd })
```

Comment (wildwestrom) [reply tooling/create-tauri-app/src/recipes/cljs.ts]: Ok. Just to be clear, I'm also gonna need `await shell('npm', ['install'], { cwd })` for the other one right?

Comment (amrbashir) [reply tooling/create-tauri-app/src/recipes/cljs.ts]: correct, only `npm` though, let me worry about `pnpm` later.

Comment (amrbashir) [reply tooling/create-tauri-app/src/recipes/cljs.ts]: there is a typo in my suggestion btw, it should be `await shell('npm', ['install'], { cwd: join(cwd, cfg.appName)})`

**Review by amrbashir — APPROVED** [review]

**Comment by wildwestrom** [comment]
I think sending another fix after approval made it so that I need another approval, or something like that. My bad.

**Review by amrbashir — APPROVED** [review]

**Comment by amrbashir** [comment]
Thanks for your contribution and I apologize for the delay.

**Pull request merged**
